### ISSUE TITLE
Breakpoints don't get hit (possibly if imported with relative paths instead of `package:`)

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#134]
1. Clone https://github.com/Dart-Code/analysis-server-typescript-generator at revision 514fd331c984a3808e3cf9a6cdcea7e1f4a66e21
1. Put a breakpoint on first line of `main` and another on first line of `ApiSpec.buildDefinitions`
1. Press `F5`; first breakpoint is hit
1. Press `F5` again; program runs to completion without hitting second breakpoint

**Comment by DanTup** [comment]
These breakpoints are being sent with different types of paths:

```
==> {"id":"5","method":"addBreakpointWithScriptUri","params":{"isolateId":"isolates/11946071",
    "scriptUri":"package:analysis_server_typescript_generator/api_spec.dart","line":21}}
==> {"id":"6","method":"addBreakpointWithScriptUri","params":{"isolateId":"isolates/11946071",
    "scriptUri":"file:///m:/Coding/Applications/analysis-server-typescript-generator/bin/main.dart","line":11}}
```

**Comment by DanTup** [comment]
The path looks good to me?

![Package breakpoint](https://cloud.githubusercontent.com/assets/1078012/18029336/4b39954a-6c8d-11e6-82ef-b7b8c21eb5fa.png)

**Comment by DanTup** [comment]
Probably as a result of the way the file is imported (see https://github.com/Dart-Code/analysis-server-typescript-generator/pull/1). However, I don't think this is a good user experience - to a user, a file is a file, regardless of how it's imported. We should aim to improve this.

Possibly this issue goes away when symlinks die, but I still don't know how that'll work.. It seems `pub` will do resolution based on the `.packages` file but we can't use `pub` for debugging (can't pass debugger/observatory flags). I'm not sure what the end goal is for executing things via `dart.exe` with no symlinks (@devoncarew - do you know; are there any docs on this? All I can find is [this](http://news.dartlang.org/2016/08/so-long-symlinks.html) but only seems to mention `pub`/`test`).

**Comment by devoncarew** [comment]
The dart vm understands `.packages` files - you can pass it an explicit reference to one, or it'll find one implicitly by looking up from the cwd. Debugging should work fine, w/ or w/o symlinks :)

**Comment by devoncarew** [comment]
(and we can definitely support purely file based paths, but self-references are pretty typical)

**Comment by DanTup** [comment]
> The dart vm understands .packages files

Ah, great! Wonder why it wasn't mentioned in that post then. I'll give it a try, I have those folders (I almost added them to the exclusion list in Code!).

> Yup, for every user breakpoint, you could set two vm breakpoints (one using the file path and one using a package: path, if one is applicable).

Sounds like a plan, should be easy to do!

> Note though that it's a very strong convention to use self-references to reach into the lib/ dir (and really enforced through different parts of the toolchain). Generally:
> 
> within a folder (lib/, bin/ test/ ...), you use relative path references

Yeah, I thought it was (which is why I did it that way first), but I'm guessing this problem will exist within folders anyway (where you say to use relatives). Even if not, I think it's good to support both if we can; it seems naff to say to a user "even though your code is valid and works, it's not valid enough for our debugger"!

The debugger is an awesome feature, it'll be good to have as few edge cases as possible when we merge it, to avoid people thinking "this is great except that it (doesn't do x properly)" and going back to their other editor's debugger! 😁

**Comment by DanTup** [comment]
Fixed by #139.
